magento/magento-cloud-docker
# MCLOUD-5857: Added Blackfire support

### Description
https://jira.corp.magento.com/browse/MCLOUD-5857: 
Added Blackfire support for magento cloud docker development by following [Blackfire for Docker docs](https://blackfire.io/docs/integrations/docker/index) and customizing it to fit our project.

### Fixed Issue

-  magento/magento-cloud-docker#131: Add Blackfire.io to Cloud Docker

### Manual testing scenarios
1. Add Blackfire configuration to `.magento.app.yaml` under `runtime.extensions` as described below.  
    ```
    runtime:
      extensions:
        - random-extension-here
        -
          name: blackfire
          configuration:
            server_id: SERVER_ID
            server_token: SERVER_TOKEN
            client_id: CLIENT_ID
            client_token: CLIENT_TOKEN
    ```
2. Generate customized (Blackfire supported) PHP images and Docker configuration as described in [docker configuration generator](https://devdocs.magento.com/cloud/reference/ece-tools-reference.html#docker-configuration-generator) and [docker development](https://devdocs.magento.com/cloud/docker/docker-development.html).
3. Add context to use locally customized PHP images as described in [Extend the Docker configuration](https://devdocs.magento.com/cloud/docker/docker-extend.html#specify-docker-build-sources).
3. Follow the rest of the guide linked [above](https://devdocs.magento.com/cloud/docker/docker-development.html) to bring up the containers and deploy Magento.
4. Ensure necessary containers are up and running.
5. Install a profiling client as described in [Blackfire documentation](https://blackfire.io/docs/up-and-running/installation#install-a-profiling-client).
6. Profile the Magento website.

### Contribution checklist
